src/main/java/com/aws/greengrass/util/platforms/unix/UnixPlatform.java

```diff
@@ -355,20 +356,51 @@ public UnixRunWithGenerator getRunWithGenerator() {
 
     @Override
     public void createUser(String user) throws IOException {
-        runCmd("useradd -r -m " + user, o -> {
-        }, "Failed to create user");
+        boolean isExist = checkIfCommandExists("useradd");
+
+        if (isExist) {
+            runCmd("useradd -r -m " + user, o -> {
+            }, "Failed to create user");
+        } else {
+            runCmd("adduser -S  " + user, l -> {
+            }, "Failed to create user");
+        }
     }
 
     @Override
     public void createGroup(String group) throws IOException {
-        runCmd("groupadd -r " + group, o -> {
-        }, "Failed to create group");
+        boolean isExist = checkIfCommandExists("groupadd");
+
+        if (isExist) {
+            runCmd("groupadd -r " + group, o -> {
+            }, "Failed to create group");
+        } else {
+            runCmd("addgroup -S " + group, l -> {
+            }, "Failed to create group");
+        }
     }
 
     @Override
     public void addUserToGroup(String user, String group) throws IOException {
-        runCmd("usermod -a -G " + group + " " + user, o -> {
-        }, "Failed to add user to group");
+        boolean isExist = checkIfCommandExists("usermod");
+
+        if (isExist) {
+            runCmd("usermod -a -G " + group + " " + user, o -> {
+            }, "Failed to add user to group");
+        } else {
+            runCmd("addgroup " + user + " " + group, l -> {
+            }, "Failed to add user to group");
+        }
+    }
+
+    /**
+     * Check if command exists.
+     *
+     * @param command command
+     * @return boolean
+     */
+    protected boolean checkIfCommandExists(String command) {
+        return UnixPlatform.getInstance().createNewProcessRunner().which(command) != null;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the resource returned by the *createNewProcessRunner* method in a try-finally block. Execution paths that do not contain closure statements were detected.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).